File: backend/app/schemas/decision.py
from datetime import datetime
from enum import Enum

from pydantic import BaseModel


class ConflictLevel(str, Enum):
    LOW = "low"
    MEDIUM = "medium"
    HIGH = "high"


class DecisionRead(BaseModel):
    id: str
    topic: str
    decision: str
    version: int
    source_chunk_id: str
    created_at: datetime

    model_config = {"from_attributes": True}


class DecisionConflictRead(BaseModel):
    id: str
    topic: str
    current_decision: str
    previous_decision: str
    level: ConflictLevel
    description: str
    current_source_chunk_id: str
    previous_meeting_id: str

    model_config = {"from_attributes": True}

File: frontend/meetingPilot-frontend/src/types/decision.ts
export enum ConflictLevel {
  LOW = 'low',
  MEDIUM = 'medium',
  HIGH = 'high',
}

export interface DecisionRead {
  id: string
  topic: string
  decision: string
  version: number
  source_chunk_id: string
  created_at: string
}

export interface DecisionConflictRead {
  id: string
  topic: string
  current_decision: string
  previous_decision: string
  level: ConflictLevel
  description: string
  current_source_chunk_id: string
  previous_meeting_id: string
}

export interface DecisionListResponse {
  meeting_id: string
  items: DecisionRead[]
}

export interface ConflictListResponse {
  meeting_id: string
  items: DecisionConflictRead[]
}
